src/main/java/com/blackducksoftware/integration/jira/config/controller/AccessConfigController.java

```diff
@@ -0,0 +1,105 @@
+import com.blackducksoftware.integration.jira.config.controller.action.AccessConfigActions;
+import com.blackducksoftware.integration.jira.config.model.BlackDuckAdminConfigSerializable;
+import com.synopsys.integration.rest.HttpMethod;
+
+@Path("/config/access")
+public class AccessConfigController extends ConfigController {
+    // This must be "package protected" to avoid synthetic access
+    final BlackDuckJiraLogger logger = new BlackDuckJiraLogger(Logger.getLogger(this.getClass().getName()));
+    final AccessConfigActions accessConfigActions;
+    final GroupPickerSearchService groupPickerSearchService;
+
+    AccessConfigController(final PluginSettingsFactory pluginSettingsFactory, final TransactionTemplate transactionTemplate, final UserManager userManager, final GroupPickerSearchService groupPickerSearchService) {
+        super(pluginSettingsFactory, transactionTemplate, userManager);
+        this.groupPickerSearchService = groupPickerSearchService;
+        this.accessConfigActions = new AccessConfigActions(getAuthorizationChecker(), groupPickerSearchService, pluginSettingsFactory);
+    }
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getPluginAdminConfiguration(@Context final HttpServletRequest request) {
+        final Object adminConfig;
+        try {
+            final boolean validAuthentication = isAuthorized(request);
+            if (!validAuthentication) {
+                return Response.status(Response.Status.UNAUTHORIZED).build();
+            }
+
+            adminConfig = executeAsTransaction(() -> accessConfigActions.getConfigWithJiraGroups(request));
+        } catch (final Exception e) {
+            return createErrorResponse(HttpMethod.GET, e);
+        }
+        return Response.ok(adminConfig).build();
+    }
+
+    @PUT
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response updateBlackDuckAdminConfiguration(final BlackDuckAdminConfigSerializable adminConfig, @Context final HttpServletRequest request) {
+        final Object responseObject;
+        try {
+            final boolean userAvailable = getAuthorizationChecker().isUserAvailable(request);
+            if (!userAvailable) {
+                return Response.status(Response.Status.UNAUTHORIZED).build();
+            }
+
+            responseObject = executeAsTransaction(() -> accessConfigActions.updateConfigWithJiraGroups(request, adminConfig.getHubJiraGroups()));
+        } catch (final Exception e) {
+            return createErrorResponse(HttpMethod.PUT, e);
+        }
+        if (responseObject != null) {
+            return Response.ok(responseObject).status(Response.Status.BAD_REQUEST).build();
+        }
+        return Response.noContent().build();
+    }
+
+    private Response createErrorResponse(final HttpMethod httpMethod, final Throwable e) {
+        final String msg = String.format("Exception performing {} : {}", httpMethod.name(), e.getMessage());
```

Comment:
Does this format work for `String.format(...)`? I thought it was only for log messages.

Reply:
For String.format(...) you need to use %s for variables

Reply:
Ya, my bad there. Already changed locally.